### PR TITLE
narrower shading

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
Narrower shading for the NIO jar. This may help with the auth issues we've been seeing for a little while.

With this change, the [repro code](https://github.com/jean-philippe-martin/nio-auth-repro) passes.

### CODE REVIEW

File: google-cloud-clients/google-cloud-contrib/google-cloud-nio/pom.xml [hunk 1]
```diff
@@ -97,20 +97,40 @@
           <shadedArtifactAttached>true</shadedArtifactAttached>
           <relocations>
             <relocation>
-              <pattern>com</pattern>
-              <shadedPattern>shaded.cloud_nio.com</shadedPattern>
+              <pattern>com.google</pattern>
```
[the remark this answers is not in the document]
Comment (jean-philippe-martin) [reply google-cloud-clients/google-cloud-contrib/google-cloud-nio/pom.xml]: Interesting, I'll try that. I hadn't seen that tag mentioned in [their documentation](https://maven.apache.org/plugins/maven-shade-plugin).

**Comment by jean-philippe-martin** [comment]
There you go, @medb !

File: google-cloud-clients/google-cloud-contrib/google-cloud-nio/pom.xml [hunk 2]
```diff
@@ -98,19 +98,33 @@
           <relocations>
             <relocation>
               <pattern>com</pattern>
-              <shadedPattern>shaded.cloud_nio.com</shadedPattern>
+              <shadedPattern>shaded.cloud_nio.com_stuff</shadedPattern>
```
Comment (medb) [inline google-cloud-clients/google-cloud-contrib/google-cloud-nio/pom.xml]: remove `_stuff` suffix?

**Comment by medb** [comment]
@jean-philippe-martin Thanks for debugging and fixing this issue!

**Comment by jean-philippe-martin** [comment]
You're welcome @medb ! I hope this will put this particular issue to bed for good!

**Comment by droazen** [comment]
@jean-philippe-martin I tested this branch using GATK just now, and it definitely appears to fix https://github.com/GoogleCloudPlatform/google-cloud-java/issues/2453! Hoping this fix can get merged and released soon!

File: google-cloud-clients/google-cloud-contrib/google-cloud-nio/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jean-philippe-martin) [reply google-cloud-clients/google-cloud-contrib/google-cloud-nio/pom.xml]: Github's great. I sent a [PR](https://github.com/apache/maven-shade-plugin/pull/4) to update the documentation.

**Comment by pongad** [comment]
AppVeyor basically never passes (I think we might as well just delete it). I'll just merge this :)

**Comment by jean-philippe-martin** [comment]
Thank you for the quick review and merge!

**Comment by jean-philippe-martin** [comment]
Found the root cause: we were hit by the "shading will also change strings in your program if they happen to match" bug (https://issues.apache.org/jira/browse/MSHADE-156). In our case the following correct code in DefaultCredentialProvider:

```
private static final String TOKEN_SERVER_ENCODED_URL = OAuth2Utils.getMetadataServerUrl() +
"/computeMetadata/v1/instance/service-accounts/default/token";
```

was incorrectly modified by the maven shade plugin:

```
private static final String TOKEN_SERVER_ENCODED_URL = OAuth2Utils.getMetadataServerUrl() +
"/shaded/cloud_nio/computeMetadata/v1/instance/service-accounts/default/token";
```

**Comment by droazen** [comment]
@jean-philippe-martin That's horrifying and bizarre...the string it clobbered didn't even resemble a package name!

**Comment by jean-philippe-martin** [comment]
Yup. But it has "com" in it!

**Pull request merged**